lms/djangoapps/courseware/user_state_client.py

```diff
@@ -149,29 +187,37 @@ def get_many(self, username, block_keys, scope=Scope.user_state, fields=None):
                 continue
 
             state = json.loads(module.state)
-            state_length += len(module.state)
+            state_length = len(module.state)
 
-            self._ddog_histogram(evt_time, 'get_many.block_size', len(module.state))
+            # NOTE: should this line be before the check for empty state?
+            self._ddog_histogram(evt_time, 'get_many.block_size', state_length)
 
             # If the state is the empty dict, then it has been deleted, and so
             # conformant UserStateClients should treat it as if it doesn't exist.
             if state == {}:
                 continue
 
+            # collect statistics for metric reporting
+            self._nr_block_count_increment('get_many', usage_key.block_type)
+            self._nr_block_size_accumulate('get_many', usage_key.block_type, state_length)
+            total_block_count += 1
```

Comment:
Maybe differentiate between blocks requested and blocks retrieved? I believe this is only capturing the latter.